src/vs/editor/contrib/anchorSelect/anchorSelect.ts

```diff
@@ -0,0 +1,172 @@
+				range: Selection.fromPositions(position, position),
+				options: {
+					stickiness: TrackedRangeStickiness.NeverGrowsWhenTypingAtEdges,
+					hoverMessage: new MarkdownString().appendText(localize('selectionAnchor', "Selection Anchor")),
+					className: 'selection-anchor'
+				}
+			}]);
+			this.decorationId = newDecorationId.length === 1 ? newDecorationId[0] : undefined;
+			this.selectionAnchorSetContextKey.set(!!this.decorationId);
+		}
+	}
+
+	goToSelectionAnchor(): void {
+		if (this.editor.hasModel() && this.decorationId) {
+			const anchorPosition = this.editor.getModel().getDecorationRange(this.decorationId);
+			if (anchorPosition) {
+				this.editor.setPosition(anchorPosition.getStartPosition());
+			}
+		}
+	}
+
+	selectFromAnchorToCursor(): void {
+		if (this.editor.hasModel() && this.decorationId) {
+			const start = this.editor.getModel().getDecorationRange(this.decorationId);
+			if (start) {
+				const end = this.editor.getPosition();
+				this.editor.setSelection(Selection.fromPositions(start.getStartPosition(), end));
+				this.cancelSelectionAnchor();
+			}
+		}
+	}
+
+	cancelSelectionAnchor(): void {
+		if (this.decorationId) {
+			this.editor.deltaDecorations([this.decorationId], []);
+			this.decorationId = undefined;
+			this.selectionAnchorSetContextKey.set(false);
+		}
+	}
+
+	dispose(): void {
+		this.cancelSelectionAnchor();
+	}
+}
+
+class SetSelectionAnchor extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.setSelectionAnchor',
+			label: localize('setSelectionAnchor', "Set Selection Anchor"),
+			alias: 'Set Selection Anchor',
+			precondition: undefined,
+			kbOpts: {
+				kbExpr: EditorContextKeys.editorTextFocus,
+				primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyCode.KEY_B),
+				weight: KeybindingWeight.EditorContrib
+			}
+		});
+	}
+
+	async run(_accessor: ServicesAccessor, editor: ICodeEditor): Promise<void> {
+		const controller = SelectionAnchorController.get(editor);
+		controller.setSelectionAnchor();
+	}
+}
+
+class GoToSelectionAnchor extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.goToSelectionAnchor',
+			label: localize('goToSelectionAnchor', "Go to Selection Anchor"),
+			alias: 'Go to Selection Anchor',
+			precondition: SelectionAnchorSet,
+		});
+	}
+
+	async run(_accessor: ServicesAccessor, editor: ICodeEditor): Promise<void> {
+		const controller = SelectionAnchorController.get(editor);
+		controller.goToSelectionAnchor();
+	}
+}
+
+class SelectFromAnchorToCursor extends EditorAction {
+	constructor() {
+		super({
+			id: 'editor.action.selectFromAnchorToCursor',
+			label: localize('selectFromAnchorToCursor', "Select from Anchor to Cursor"),
+			alias: 'Select from Anchor to Cursor',
+			precondition: SelectionAnchorSet,
+			kbOpts: {
+				kbExpr: EditorContextKeys.editorTextFocus,
+				primary: KeyChord(KeyMod.CtrlCmd | KeyCode.KEY_K, KeyCode.KEY_K),
```

Comment:
We should look if `cmd+k cmd+k` is available. You can check the availability across OSes here -- https://github.com/microsoft/vscode-docs/tree/master/build/keybindings